packages/webapp/src/components/ui/PaginatedList/PaginatedList.tsx

```diff
@@ -93,6 +101,79 @@ export const PaginatedList = memo(function PaginatedList<T>({
 		},
 		[onSearchValueChange]
 	)
+
+	const [isListSorted, setListSorted] = useState<boolean>(false)
+	const [listSortingInfo, setListSortingInfo] = useState<ListSorting>({
+		key: columns?.[0]?.key ?? null,
+		order: Sorting.Order.ASC,
+		sortingValue: nullFn,
+		sortingFunction: nullFn
+	})
+
+	// List sorted based on user selected Header column and ASC/DESC order.
+	const sortedList = !isListSorted
+		? list
+		: [...list].sort((itemA, itemB) => {
+				return listSortingInfo.sortingFunction(
+					listSortingInfo.sortingValue(itemA),
+					listSortingInfo.sortingValue(itemB),
+					listSortingInfo.order
+				)
+		  })
+
+	/*
+		Set sorting info based on state of the header column.
+		If a different Header column is selected, it's set to ASC and 
+		remove the sorting from the previous Header column.
+	 */
+	const handleHeaderClick: OnHeaderClick = (headerKey: string) => {
```

Comment:
You may consider wrapping this in useCallback to prevent re-rendering

Reply:
I don't understand why this would be useful, as the goal is to re-render the whole list sorted?

I gave it a try, and it breaks everything, because I modify the `columns` props in my callback:
```TSX
 const memoizedHeaderClick = useCallback(
     (headerKey: string) => {
            handleHeaderClick(headerKey)
     },
     [columns]
)
```

Reply:
I usually memoize all callbacks being passed down to JSX - otherwise the child components will render on every cycle because the callback function identity is changing on every render. It looks like you're likely missing some hook dependencies is useCallback. If this is not feasible for now, I wouldn't sweat it